aws static website component yaml blog post

sean1588 opened the pull request:
The blog post for AWS static website.

sean1588 commented on themes/default/content/blog/aws-static-website-component/index.md:
```diff
@@ -0,0 +1,193 @@
+meta_desc: Learn how to easily deploy your static website using the AWS Static Website Component. 
+meta_image: "" 
+authors:
+    - sean-holung
```
temporarily put me here, since we do not yet have roberta as an author in our system and is causing errors

zchase commented:
@sean1588 My suggestion for a title would be, "Deploy a React App to AWS in 11 lines of YAML".

Also would suggest to do React instead of Hugo.

cnunciato commented on themes/default/content/blog/aws-static-website-component/index.md:
```diff
@@ -0,0 +1,188 @@
+
+The AWS Static Website has several input properties. Here are three of the most commonly used.
+
+* **sitePath** - the root directory that contains the website’s contents (required).
+* **withCDN** - sets a flag to provision a CloudFront CDN.
+* **targetDomain** - the domain that serves the content. A Route53 hosted zone must exist for this domain if you specify this option.
```
```suggestion
* `sitePath` - the root directory that contains the website’s contents (required).
* `withCDN` - sets a flag to provision a CloudFront CDN.
* `targetDomain` - the domain that serves the content. A Route53 hosted zone must exist for this domain if you specify this option.
```

cnunciato commented on themes/default/content/blog/aws-static-website-component/index.md:
```diff
@@ -0,0 +1,188 @@
+
+## Output Properties
+
+The AWS Static Website has several output properties. The one you’ll probably use the most is **websiteURL**, which is the URL to access the website. See the component’s [reference page](https://www.pulumi.com/registry/packages/aws-static-website/) for the others.
```
```suggestion
The AWS Static Website component has several output properties. The one you’ll probably use the most is `websiteURL`, which is the URL to access the website. See the component’s [reference page]({{< relref "/registry/packages/aws-static-website" >}}) for the others.
```

cnunciato commented on themes/default/content/blog/aws-static-website-component/index.md:
```diff
@@ -0,0 +1,188 @@
+* **withCDN** - sets a flag to provision a CloudFront CDN.
+* **targetDomain** - the domain that serves the content. A Route53 hosted zone must exist for this domain if you specify this option.
+
+See the component’s [reference page](https://www.pulumi.com/registry/packages/aws-static-website/) for the others.
```
```suggestion
See the component’s [reference page]({{< relref "/registry/packages/aws-static-website" >}}) for the others.
```

cnunciato commented on themes/default/content/blog/aws-static-website-component/index.md:
```diff
@@ -0,0 +1,188 @@
+* You can examine the [GitHub repository](https://github.com/pulumi/pulumi-aws-static-website) that contains the code for the component.
+* If you want to quickly try out the component, this [GitHub repo](https://github.com/sean1588/create-react-app-pulumi) houses a complete React example project that you can clone.
+* If you want to learn about other Pulumi components, take a look at the [registry page](https://pulumi.com/registry), which catalogs them all.
+* If you want to delve deeper into Pulumi, the [Getting Started](https://www.pulumi.com/docs/get-started/) guide is a good first step.
+* If you’re interested in learning more about Pulumi concepts, try the [Architecture & Concepts](https://www.pulumi.com/docs/intro/concepts/) page.
```
```suggestion
* If you want to delve deeper into Pulumi, the [Getting Started]({{< relref "/docs/get-started"  >}}) guide is a good first step.
* If you’re interested in learning more about Pulumi concepts, try the [Architecture & Concepts]({{< relref "/docs/intro/concepts" >}}) page.
```

cnunciato commented on themes/default/content/blog/aws-static-website-component/index.md:
```diff
@@ -0,0 +1,188 @@
+
+## Install Pulumi
+
+To install Pulumi for AWS, go to [Before You Begin](https://www.pulumi.com/docs/get-started/aws/begin/). There are instructions for macOS, Windows and Linux. You need to have an AWS account.
```
```suggestion
To install Pulumi for AWS, go to [Before You Begin]({{< relref "/docs/get-started/aws/begin" >}}). There are instructions for macOS, Windows and Linux. You need to have an AWS account.
```

cnunciato commented on themes/default/content/blog/aws-static-website-component/index.md:
```diff
@@ -0,0 +1,188 @@
+
+## Input Properties
+
+The AWS Static Website has several input properties. Here are three of the most commonly used.
```
```suggestion
The AWS Static Website component has several input properties. Here are three of the most commonly used.
```

cnunciato commented on themes/default/content/blog/aws-static-website-component/index.md:
```diff
@@ -0,0 +1,188 @@
+  web:
+    type: "aws-static-website:index:Website"
+    properties:
+      sitePath: "./public"
```
```suggestion
      sitePath: "./my-website/build"
```

cnunciato commented on themes/default/content/blog/aws-static-website-component/index.md:
````diff
@@ -0,0 +1,188 @@
+  websiteURL: ${web.websiteURL}
+```
+
+All you have to do is add the line: **withCDN: true** and the component sets up the CDN instance for you. Again, run `pulumi up`, confirm that you want to proceed, and Pulumi deploys the website. Here’s an example of what you’ll see.
````
```suggestion
All you have to do is add the line `withCDN: true` and the component sets up the CDN instance for you. Again, run `pulumi up`, confirm that you want to proceed, and Pulumi deploys the website. Here’s an example of what you’ll see.
```

cnunciato reviewed (commented):
Nice! This is coming along well. Sorry to bombard you with comments! I've tried to include as many concrete suggestions as I could to make it easy to get through.

sean1588 commented on themes/default/content/blog/aws-static-website-component/index.md:
```diff
@@ -0,0 +1,188 @@
+    - yaml
+    - configuration
+    - components
+    - website
```
cool, I was wondering if there was a fixed set of tags.

sean1588 commented on themes/default/content/blog/aws-static-website-component/index.md:
```diff
@@ -0,0 +1,188 @@
+
+## What the Component Does
+
+As we said, the component deploys a static website to AWS and, optionally, a CloudFront CDN. You simply add a few lines of code, set a few configuration parameters, run `pulumi up` and let the component do the rest. The component sets up:
```
agreed this sounds better.

sean1588 commented on themes/default/content/blog/aws-static-website-component/index.md:
```diff
@@ -0,0 +1,188 @@
+
+## Some Examples
+
+Let’s look at two examples of how to use the component. You’ll find both examples and more at [https://gist.github.com/sean1588/f6ca2e09550e53c869259d569f29e9a1](https://gist.github.com/sean1588/f6ca2e09550e53c869259d569f29e9a1). We’re going to use YAML in this post but the gist also contains the JSON equivalents.
```
ahh yes, I think we can get rid of this. Roberta had added this to the post. The gist wasa really meant just to be for her reference when writing and said I would link a repo with examples when we publish. thanks!

sean1588 commented on themes/default/content/blog/aws-static-website-component/index.md:
```diff
@@ -0,0 +1,188 @@
+  web:
+    type: "aws-static-website:index:Website"
+    properties:
+      sitePath: "../build"
```
ahhh.... yes good catch. yeah you are correct. public is what hugo produced.

sean1588 commented on themes/default/content/blog/aws-static-website-component/index.md:
````diff
@@ -0,0 +1,188 @@
+│   ├── Pulumi.yaml
+```
+
+**Note:** Pulumi will add some other files to the **infrastructure** directory but you can ignore them.
````
I definitely agree and did mention this to Roberta during the review and she felt it would be best to word it this way as to not distract the user and not need to go into an explanation of stacks. I'll see if I can come up with a rewording of this, because I also feel we shouldn't be saying to just ignore the files.

sean1588 commented on themes/default/content/blog/aws-static-website-component/index.md:
```diff
@@ -0,0 +1,188 @@
+
+![hugo-example-simple](react-example-simple.png)
+
+If you click **View Live**, you will go to the Pulumi Console, where you can see all of the information available about a deployment and explore the resources that were deployed. For example, you can see a resource graph and a list of resources. Here’s an example of a resource graph.
```
good to know that we have an official way of referencing it now!

sean1588 commented on themes/default/content/blog/aws-static-website-component/index.md:
```diff
@@ -0,0 +1,188 @@
+title: Deploy a React App to AWS using 11 lines of YAML
+date: 2022-06-27
+meta_desc: Learn how to easily deploy your static website using the AWS Static Website Component. 
+meta_image: "" 
```
OK, is this like a preview image that gets shown on the site?

cnunciato replied:
Yes. You’ll see examples in other blog posts, usually as a file named meta.png. https://github.com/pulumi/pulumi-hugo/blob/master/BLOGGING.md#social-meta-images

sean1588 commented:
> Sorry to bombard you with comments! I've tried to include as many concrete suggestions as I could to make it easy to get through.

I appreciate that @cnunciato. Thank you and sorry for all the relref misses. I am sure it is also time consuming for you to add the suggestions. If it happens again, you can just leave a comment as I don't want to make this time consuming for you also. I appreciate it though!

sean1588 commented:
I updated the meta image (thanks christian for creating it!), and also updated the title. I also added some changes to include keywords in the blog content that Sara had suggested.

cnunciato commented:
Sorry I didn’t catch this until now — I can take a quick peek this weekend for ya.

cnunciato reviewed (approved):
One little spacing thing, otherwise I think we can ship!

sean1588 commented on themes/default/content/blog/deploy-aws-static-website-with-yaml/index.md:
```diff
@@ -0,0 +1,207 @@
+* An Amazon S3 bucket
+* An Amazon CloudFront CDN (optional)
+    * An Amazon Route 53 DNS record
+    * An Amazon Certificate Manager
```
I like.
